### DISCUSSION TITLE
How to define command to send value without read?

### DISCUSSION THREAD

**Discussion by PeterPirog** [pymeasure/pymeasure#475] [Q&A] [answered]
I would like to add zeroing function for  HP 34401 multimeter, but I'm not sure how to define it:

`zeroing=Instrument.control("CALC:FUNC NULL;:CALC:STAT ON", "Zeroing")`

I will be grateful for any suggestions.

Peter

**Comment by LongnoseRob** [comment, marked as answer]
If you have a SCPI command that cannot be queried, I woudl recommend to use [Instrument.setting](https://pymeasure.readthedocs.io/en/latest/api/instruments/instruments.html#pymeasure.instruments.Instrument.setting)

Another idea would be to implement the CALC:FUNC with all possible values as one propery and add another propery for the CALC:STATE.

Do you also would like to set the null-value(offset) by remote?

**Comment by PeterPirog** [comment]
@LongnoseRob Thank You fot the answer. Yes  I would like to use null value by remote to measure resistance change.

I try to understand how to define new instruments because I have some Fluke multifunction calibrators and RF power meters  which I would like to add.

**Comment by bilderbuchi** [comment]
Indeed, a setting is the way to go here. I'll convert this into a discussion, where it is more appropriate, to keep the issue tracker "clean" 😁

**Comment by PeterPirog** [comment]
I did something like that, but there is no effect:

```
from pymeasure.instruments import Instrument

class HP34401A(Instrument):
    """ Represents the HP 34401A instrument.
    """
    id = Instrument.measurement("*IDN?", "Reads the instrument identification")
    voltage_dc = Instrument.measurement("MEAS:VOLT:DC? DEF,DEF", "DC voltage, in Volts")
    
    voltage_ac = Instrument.measurement("MEAS:VOLT:AC? DEF,DEF", "AC voltage, in Volts")
    
    current_dc = Instrument.measurement("MEAS:CURR:DC? DEF,DEF", "DC current, in Amps")
    
    current_ac = Instrument.measurement("MEAS:CURR:AC? DEF,DEF", "AC current, in Amps")
    
    resistance = Instrument.measurement("MEAS:RES? DEF,DEF", "Resistance, in Ohms")
    
    resistance_4w = Instrument.measurement("MEAS:FRES? DEF,DEF", "Four-wires (remote sensing) resistance, in Ohms")


    def __init__(self, resourceName, **kwargs):
        super(HP34401A, self).__init__(
            resourceName,
            "HP 34401A",
            **kwargs
        )
    def zeroing(self):
        self.setting("CALC:FUNC NULL;:CALC:STAT ON", "Zeroing")
```

**Reply by bilderbuchi** [reply]
I'm not familiar with the device, but two possible issues:
* not sure about the command syntax, but you have `"CALC:FUNC NULL;:CALC:STAT ON"`, is the colon after the semicolon permissible? should it not be `"CALC:FUNC NULL;CALC:STAT ON"`?
* According to the manual, there is also `CALCulate:NULL:OFFSet {<value>|MINimum|MAXimum}`, which seems to make the "null function" set some register (before it was only "activated", whatever that means).
